examples/tensorflow-mnist-pixel-threshold/demo.ipynb

```diff
@@ -182,6 +182,72 @@
     "restapi_client.list_queues()"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+      "This example also makes use of the `evaluation` custom task plugin package stored locally under the `task-plugins/securingai_custom/evaluation` directory.\n",
+    "To register these custom task plugins, we first need to package them up into an archive.\n",
+    "For convenience, the `Makefile` provides a rule for creating the custom task plugins archive file, just run `make custom-plugins`,"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Create the workflows.tar.gz file\n",
+    "make custom-plugins"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that the `evaluation` custom task plugin package is packaged into an archive file, next we register it by uploading the file to the REST API.\n",
+    "Note that we need to provide the name to use for custom task plugin package, this name must be unique under the custom task plugins namespace.\n",
+    "For a full list of the custom task plugins, use `restapi_client.restapi_client.list_custom_task_plugins()`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "response_custom_plugins = restapi_client.get_custom_task_plugin(name=\"evaluation\")\n",
+    "#restapi_client.delete_custom_task_plugin(name=\"evaluation\")\n",
+    "if response_custom_plugins is None or \"Not Found\" in response_custom_plugins.get(\"message\", []):\n",
+    "    response_custom_plugins = restapi_client.upload_custom_plugin_package(\n",
+    "        custom_plugin_name=\"evaluation\",\n",
+    "        custom_plugin_file=CUSTOM_PLUGINS_EVALUATION_TAR_GZ,\n",
+    "    )\n",
+    "\n",
+    "response_custom_plugins"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If at any point you need to update one or more files within the `evaluation` plugin package, you will need to unregister/delete the custom task plugin first using the REST API.\n",
+    "This can be done as follows,"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "# Delete the 'evaluation' custom task plugin package\n",
+    "restapi_client.delete_custom_task_plugin(name=\"evaluation\")\n",
+    "```"
+   ]
+  },
+  
```

Comment:
This example adds a new file, [pixelthreshold.py](https://github.com/usnistgov/dioptra/tree/2b722e84670251bcaf6e5f3167558196801cc3fb/examples/tensorflow-mnist-pixel-threshold/task-plugins/securingai_custom/evaluation), into the `task-plugins/securingai_custom/evaluation/` folder, and registers this under the "evaluation" custom task plugin namespace. This will lead to a naming conflict if a user tries to register [the tensorflow-mnist-classifier example's "evaluation" custom task plugin](https://github.com/usnistgov/dioptra/tree/29ee50438a001715d1ca0f3038527fe286e74719/examples/tensorflow-mnist-classifier/task-plugins/securingai_custom/evaluation) and this one on the same deployment.

How to fix:

1. Move `task-plugins/securingai_custom/evaluation/pixelthreshold.py` into a new folder, such as `task-plugins/securingai_custom/pixel_threshold/pixelthreshold.py`. You'll also want to add an `__init__.py` into this folder. The `evaluation/` folder can remain with the `import_keras.py` and `tensorflow.py` files since those are identical to their [tensorflow-mnist-classifier counterpats](https://github.com/usnistgov/dioptra/tree/29ee50438a001715d1ca0f3038527fe286e74719/examples/tensorflow-mnist-classifier/task-plugins/securingai_custom/evaluation).
2. In `demo.ipynb`, extend the notebook to register two task plugins, "evaluation" and the new one created in step 1 ("pixel_threshold", for example). This can be done by duplicating the code cells using `restapi_client.upload_custom_plugin_package()` and updating them to register the "pixel_threshold" custom plugins. The "evaluation" plugins will upload if the user hasn't tried the tensorflow-mnist-classifier example yet. The surrounding explanatory text will need to be updated to reflect these additions to the notebook.
3. Update your scripts in the `src` directory to use the new custom task plugin name where appropriate, for example https://github.com/usnistgov/dioptra/blob/2b722e84670251bcaf6e5f3167558196801cc3fb/examples/tensorflow-mnist-pixel-threshold/src/pt.py#L255 would potentially change to `f"{_CUSTOM_PLUGINS_IMPORT_PATH}.pixel_threshold"` (it should match the `custom_plugin_name` chosen in step 2).
4. Extend the Makefile to package up the "pixel_threshold" custom task plugin separately. This can largely be done by duplicating the variables and rules containing `EVALUATION_` and `evaluation` in their names and editing them to match the new custom task plugin. For example, you could duplicate https://github.com/usnistgov/dioptra/blob/2b722e84670251bcaf6e5f3167558196801cc3fb/examples/tensorflow-mnist-pixel-threshold/Makefile#L257-L259 and edit it as follows:
   ```makefile
   $(CUSTOM_TASK_PLUGINS_PIXEL_THRESHOLD_TARBALL_FILE): $(CUSTOM_PLUGIN_PIXEL_THRESHOLD_SRC_FILES)
   	$(call make_tarball,$@,$(addprefix pixel_threshold/,$(notdir $(CUSTOM_PLUGIN_PIXEL_THRESHOLD_SRC_FILES))),-C $(PLUGINS_CUSTOM_DIR))
   	$(call chmod,644,$(CUSTOM_TASK_PLUGINS_PIXEL_THRESHOLD_TARBALL_FILE))
   ```
   This section of the Makefile rules could then be extended https://github.com/usnistgov/dioptra/blob/2b722e84670251bcaf6e5f3167558196801cc3fb/examples/tensorflow-mnist-pixel-threshold/Makefile#L218-L222
   to
   ```makefile
   ## Create custom plugins tarballs
   custom-plugins: custom-plugins-evaluation custom-plugins-pixel-threshold

   ## Create custom evaluation plugins tarball
   custom-plugins-evaluation: $(CUSTOM_TASK_PLUGINS_EVALUATION_TARBALL_FILE)

   ## Create custom pixel threshold plugins tarball
   custom-plugins-pixel-threshold: $(CUSTOM_TASK_PLUGINS_PIXEL_THRESHOLD_TARBALL_FILE)
   ```
   Other variables in the Makefile will also need to be duplicated and edited in the same manner.

I believe those are the main things that would need to be updated to handle the second custom task plugin folder.